noir/bootstrap.sh

```diff
@@ -31,19 +31,31 @@ function noir_sync {
 # Calculate the content hash for caching, taking into account that `noir-repo`
 # is not part of the `aztec-packages` repo itself, so the `git ls-tree` used
 # by `cache_content_hash` would not take those files into account.
+function noir_repo_content_hash {
+  echo $(REPO_PATH=./noir-repo AZTEC_CACHE_COMMIT=HEAD cache_content_hash $@)
+}
+
+# Get the cache content hash. It should only be based on files committed to `aztec-packages`
+# in order to be able to support using `AZTEC_CACHE_COMMIT` for historical queries.
 function noir_content_hash {
-  function noir_repo_content_hash {
-    echo $(REPO_PATH=./noir-repo cache_content_hash $@)
-  }
-  with_tests=${1:-0}
+  # Currently we don't make a distinction between test and non-test hash
+  tests=${1:-0}
+
+  # If there are changes in the noir-repo which aren't just due to the patch applied to it,
+  # then just disable the cache. We could have workarounds, such as committing the content
+  # hash that we can calculate with `noir_repo_content_hash` to files, but it's considered
+  # gross. The tradeoff is that we rebuild based on *any* change captured in the `noir-repo-ref`,
+  # and that if we add new commits to a feature branch in `noir-repo` then we'd need to return
+  # the actual content hash if we wanted the build to use caching.
   noir_hash=$(cache_content_hash .rebuild_patterns)
-  noir_repo_hash=$(noir_repo_content_hash .noir-repo.rebuild_patterns)
-  if [ "$with_tests" == "1" ]; then
-    noir_repo_hash_tests=$(noir_repo_content_hash .noir-repo.rebuild_patterns_tests)
+
+  if [ -f .noir-repo.force-cache ]; then
```

Comment:
nit: can be in .gitignore

Reply:
To make it local-only? I thought we could send it to CI as well, with the discipline to remove it before the finalising the PR. But yeah.

Reply:
Otherwise we could use an env var

Reply:
Removed this file: now we use `noir_repo_content_hash` _iff_ we're on a branch, and there are no uncommitted changes. This way I can do my feature branch work without affecting your scripts I hope, because the feature branch will never be committed to the `master` branch on `aztec-packages`.